core/endorser/endorser.go

```diff
@@ -441,6 +449,30 @@ func (e *Endorser) ProcessProposalSuccessfullyOrError(up *UnpackedProposal) (*pb
 	}, nil
 }
 
+// performSimulationWithThrottling simulates the proposal and returns endorsement info and simulation result.
+// It throttles concurrent calls to limit the number of os threads created due to concurrent ledger access.
+func (e *Endorser) performSimulationWithThrottling(ctx context.Context, up *UnpackedProposal, txParams *ccprovider.TransactionParams) (
+	*lifecycle.ChaincodeEndorsementInfo, *pb.Response, []byte, *pb.ChaincodeEvent, error) {
+
+	if err := e.ThrottleSemaphore.Acquire(ctx); err != nil {
```

Comment:
Just like I commented in the deliver service code, can you verify that this ctx.Done is already not getting called in a separate goroutine already in the caller path.

Reply:
Once the code path gets to endorser.ProcessProposal, other than the semaphore, ctx is only used to extract context value which doesn't call ctx.Done. The grpc layer calls ctx.Done in DialContext, but it will return error before getting into endorser.